packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -23,6 +23,58 @@ export 'package:flutter/rendering.dart' show RelativeRect;
 /// [AnimatedWidget] is most useful for widgets that are otherwise stateless. To
 /// use [AnimatedWidget], simply subclass it and implement the build function.
 ///
+///{@tool sample}
+///
+/// This code defines a widget called `Spinner` that spins a green square
+/// continually. It is built with an [AnimatedWidget].
+///
+/// ```dart
+/// class Spinner extends StatefulWidget {
+///   @override
+///   _SpinnerState createState() => _SpinnerState();
+/// }
+///
+/// class _SpinnerState extends State<Spinner> with TickerProviderStateMixin {
+///   AnimationController _controller;
+///
+///   @override
+///   void initState() {
+///     super.initState();
+///     _controller = AnimationController(
+///       duration: const Duration(seconds: 10),
+///       vsync: this,
+///     )..repeat();
+///   }
+///
+///   @override
+///   void dispose() {
+///     _controller.dispose();
+///     super.dispose();
+///   }
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return AnimatedContainer(controller: _controller);
+///   }
+/// }
+///
+/// class AnimatedContainer extends AnimatedWidget {
+///   AnimatedContainer({Key key, AnimationController controller})
+///       : controller = controller,
+///         super(key: key, listenable: controller);
+///   final AnimationController controller;
```

Comment:
In the framework instead of holding on to the controller, we actually just cast `listenable` to the appropriate type, e.g.:

* https://github.com/flutter/flutter/blob/c37b7c535c807f030884a4e02c67b3d1e0105bf3/packages/flutter/lib/src/widgets/modal_barrier.dart#L140
* https://github.com/flutter/flutter/blob/6d134e0c866ce90a4dc7a02af6a8f1079c2a4fdc/examples/flutter_gallery/lib/gallery/backdrop.dart#L90

Not sure what the more idiomatic usage is that we'd want to show in the sample. /cc @HansMuller

Reply:
Thinking about this a little more, I actually prefer the style we use in the framework since it doesn't sparkle any confusion weather the `controller` instance of the subclass is different from the `listenable` instance of the super class.

Reply:
@goderbauer I agree, the code is a little clearer for not introducing a field.

Reply:
I updated the code to use a getter, following the same pattern from modal_barrier

Reply:
Updated to use a getter with the parent's listenable property

Reply:
Oh, and changed construct to const